api/src/main/java/org/openmrs/validator/CohortValidator.java

```diff
@@ -41,13 +42,18 @@ public void validate(Object obj, Errors errors) {
 
 
 		Cohort cohort = (Cohort) obj;
-		Collection<CohortMembership> members = cohort.getMembers();
-		if (!CollectionUtils.isEmpty(members)) {
-			for (CohortMembership member : members) {
-				Patient p = member.getPatient();
-				if (p.getVoided() && !member.getVoided()) {
-					String eMessage = "Patient " + p.getPatientId() + " is voided, cannot add voided members to a cohort";
-					errors.rejectValue("members", "Cohort.patientAndMemberShouldBeVoided", eMessage);
+		if (!cohort.getVoided()) {
+			Collection<CohortMembership> members = cohort.getMemberships();
+			if (!CollectionUtils.isEmpty(members)) {
+				for (CohortMembership member : members) {
+					Patient p = member.getPatient();
+					int dateCompare = OpenmrsUtil.compareWithNullAsLatest(member.getStartDate(), member.getEndDate());
+					if ((p.getVoided() && !member.getVoided())
+							|| (member.getStartDate() == null || dateCompare == 1)) {
```

Comment:
Do you realize that if startDate is null based on the method you called above then the dateCompare value would be 1? Meaning the `member.getStartDate() == null` check is redundant